### ISSUE TITLE
New command Proposal: Test-DbaKerberos

### ISSUE THREAD

**Issue by potatoqualitee** [sqlcollaborative/dbatools#6229]
* Run Test-DbaSpn
* Compare system clocks - current to remote
* Test for duplicates (`setspn -X`)
* Something with klist?
* Check DNS records
* Check to ensure the account is not locked out
* Check for CNAMES
* Check for date of service account creation
* Check hosts file
* tcp/88, udp/88 (kerberos authentication) and tcp/389, udp/389 from source server to the domain controllers is enabled
* Local security pol. setting: impersonate client after authentication
* Test-ComputerSecureChannel
* Ensure RC4_HMAC_MD5 exists in the policy "Network Security: Configure encryption types allowed for Kerberos".
* Ensure account is not denied login at Windows level (local security policy)

https://www.sqlserver-dba.com/2012/03/sql-server-kerberos-and-krb_ap_err_modified-.html

https://social.msdn.microsoft.com/Forums/en-US/f52efc25-257c-4d9f-9472-b052d466aea8/sql-server-management-studio-2014-without-quotrc4hmacmd5quot?forum=sqlsecurity

Anything I missed from all these suggestions: https://twitter.com/cl/status/1202740479068114944
https://twitter.com/cl/status/1202257822265810944

Basically, many of the steps from here: https://help.deepsecurity.trendmicro.com/11_2/on-premise/kerberos-mssql.html and potentially other blogs.

**Comment by potatoqualitee** [comment]
start a job that executes some sqlclient sqlconnection so that it's a fresh connect. i found old windows can retain their broken kerb even after kerb has been fixed

**Comment by wsmelton** [comment]
Klist, we'd need to test OS version. It is only on 2012 or higher...

Need to also validate AD permissions or just make sure try catch is in place.

**Comment by potatoqualitee** [comment]
klist is available on my test win7 box 🤔

**Comment by potatoqualitee** [comment]
@claude ive always wanted this command! please research and make a long ass checklist of things we should check

**Comment by potatoqualitee** [comment]
@claude how many of those are reasonable to add to `Test-DbaKerberos`

**Comment by potatoqualitee** [comment]
@claude pls create command for tier 1 and 2. note that appveyor wont support kerberos so??? how to handle

**Comment by potatoqualitee** [comment]
@claude please proceed, your quota has just been upped

**Comment by potatoqualitee** [comment]
@claude please proceed, your quota has just been upped

**Comment by potatoqualitee** [comment]
Fixed 👍🏼
